### DISCUSSION TITLE
Custom ValidationError messages

### DISCUSSION THREAD

**Discussion by wsaca** [Netflix/dgs-framework#135] [Q&A]
Hi, is there any way to customize "ValidationError" messages?

These are the posible error types returned when the validation rules are applied:
https://github.com/graphql-java/graphql-java/blob/v16.2/src/main/java/graphql/validation/ValidationErrorType.java

**Comment by jmbarroso** [comment]
Hi,

I haven't got to implement error handling but I recommend reading this article from the doc that may be helpful: https://netflix.github.io/dgs/error-handling/

**Reply by srinivasankavitha** [reply]
Thanks @jmbarroso. That is exactly the link I was going to suggest, mainly the custom exception handling.

**Reply by wsaca** [reply]
@jmbarroso thanks, but instead of customize the data fetcher exceptions I need to customize the validation errors created when the query is validated, this happens before the data fetcher is executed.

**Reply by srinivasankavitha** [reply]
For customizing behavior before/after execution of data fetchers, you can try using the instrumentation class. Extending SimpleInstrumentation should give you hooks to implement custom error handling etc. Here is a doc that describes doing so for adding metrics etc, but the concept is similar.
https://netflix.github.io/dgs/advanced/instrumentation/

Not sure if this gets you the ability to customize validation errors though.

**Reply by wsaca** [reply]
@srinivasankavitha I think it could work, I will try it...
